src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -889,6 +889,19 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(CSharpSyntax
 
             Debug.Assert(members.Count > 0);
 
+            if (!hasErrors && members[0] is LocalFunctionSymbol)
+            {
+                Debug.Assert(members.Count == 1 && members[0].Locations.Length == 1);
```

Comment:
I don't think so. Right now with the shadowing rule, if a single member is a LocalFunctionSymbol, the assert follows, because in LocalScopeBinder.cs LookupSymbolsInSingleBinder does a LookupResult.MergeEqual passing in a SingleLookupResult, so I don't think that's possible to have multiple results.